File: api/src/endpoint/question/create_question_comment.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::question::CommentResource};

pub struct CreateQuestionComment;

impl Endpoint for CreateQuestionComment {
    const PATH: &'static str = "/repository/{owner}/{repo}/question/{number}/comment";
    const METHOD: http::Method = http::Method::POST;

    type Request = CreateQuestionCommentRequest;
    type Response = CreateQuestionCommentResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct CreateQuestionCommentRequest {
    pub body: String,
}

pub type CreateQuestionCommentResponse = CommentResource;

File: gitdot-api-ts/src/endpoint/question/create-question-comment.ts
import { z } from "zod";

import { CommentResource } from "../../resource";

export const CreateQuestionCommentRequest = z.object({
  body: z.string(),
});
export type CreateQuestionCommentRequest = z.infer<
  typeof CreateQuestionCommentRequest
>;

export const CreateQuestionComment = {
  path: "/repository/{owner}/{repo}/question/{number}/comment",
  method: "POST",
  request: CreateQuestionCommentRequest,
  response: CommentResource,
} as const;
export type CreateQuestionComment = typeof CreateQuestionComment;
